model-archiver/model_archiver/tests/integ_tests/test_integration_model_archiver.py

```diff
@@ -0,0 +1,162 @@
+    assert manifest.get("model").get("handler") == test.get("handler")
+
+
+def validate_files(file_list, prefix, regular):
+    assert os.path.join(prefix, MANIFEST_FILE) in file_list
+    assert os.path.join(prefix, "service.py") in file_list
+
+    if regular:
+        assert os.path.join(prefix, "dummy-artifacts.txt") in file_list
+        assert os.path.join(prefix, "dir/1.py") in file_list
+    else:
+        assert os.path.join(prefix, "model.onnx") in file_list
+
+
+def validate_tar_archive(test_cfg):
+    import tarfile
+    file_name = os.path.join(test_cfg.get("exportPath"), test_cfg.get("modelName") + ".tar.gz")
+    f = tarfile.open(file_name, "r:gz")
+    manifest = json.loads(f.extractfile(os.path.join(test_cfg.get("modelName"), MANIFEST_FILE)).read())
+    validate_manifest_file(manifest, test_cfg)
+    validate_files(f.getnames(), test_cfg.get("modelName"), "regular_model" in test_cfg.get("modelPath"))
+
+
+def validate_noarchive_archive(test):
+    file_name = os.path.join(test.get("exportPath"), test.get("modelName"), MANIFEST_FILE)
+    manifest = json.loads(open(file_name).read())
+    validate_manifest_file(manifest, test)
+
+
+def validate_mar_archive(test):
+    import zipfile
+    file_name = os.path.join(test.get("exportPath"), test.get("modelName") + ".mar")
+    zf = zipfile.ZipFile(file_name, "r")
+    manifest = json.loads(zf.open(MANIFEST_FILE).read())
+    validate_manifest_file(manifest, test)
+
+
+def validate_archive_content(test):
+    fmt = test.get("archiveFormat")
+    if fmt == "tgz":
+        validate_tar_archive(test)
+    if fmt == "no-archive":
+        validate_noarchive_archive(test)
+    if fmt == "default":
+        validate_mar_archive(test)
+
+
+def validate(test):
+    validate_archive_exists(test)
+    validate_archive_content(test)
+
+
+def test_model_archiver():
+    
+    f = open("model_archiver/tests/integ_tests/configuration.json", "r")
+    tests = json.loads(f.read())
+    for t in tests:
+        try:
+            delete_file_path(t.get("exportPath"))
+            create_file_path(t.get("exportPath"))
+            t = update_tests(t)
+            cmd = "model-archiver " \
+                  "--model-name {} " \
+                  "--model-path {} " \
+                  "--handler {} " \
+                  "--runtime {} " \
+                  "--export-path {} " \
+                  "--archive-format {}".format(t.get("modelName"),
+                                               t.get("modelPath"),
+                                               t.get("handler"),
+                                               t.get("runtime"),
+                                               t.get("exportPath"),
+                                               t.get("archiveFormat"))
+            if t.get("force"):
+                cmd += " -f"
+
+            # TODO: Add tests to check for "convert" functionality
+            if run_test(t, cmd):
```

Comment:
What would happen if run_test failed with a return of 0 in the method ? This should stop the execution of the integ test suite ? Or continue silently ?

Reply:
Tests will stop at first failure.

Reply:
BTW, return 0 is for tests that are expected to fail. In these cases there are no artifacts to validate since the model-archiver throws exception.

Reply:
Aah. I see.